### PR TITLE
JSON Schema Normalisation

### PR DESCRIPTION
**Pull request by XVincentX**
1. Moves the description in the JSON Schema
2. Removes the `ISchema` interface in favour of regular JSONSchema

### CODE REVIEW

File: src/http-spec.ts [hunk 1]
```diff
@@ -73,9 +73,8 @@ export interface IHttpParam {
   name: string;
   style: HttpParamStyles;
   content?: IHttpContent;
-  description?: string;
```
Comment (marbemac) [inline src/http-spec.ts]: I think we can actually leave description as is (see https://gist.github.com/marbemac/1cd45ca27846f2b838a3ab369abc32cd#file-httpop-json-L35).

Comment (XVincentX) [reply src/http-spec.ts]: Restored. The schema could still have a description though, is it ok to have two of them?

**Comment by XVincentX** [comment]
@marc I've modified the interface `IHttpParam` to extend the `IHttpContent` and restored the description.

**Comment by marbemac** [comment]
Can you pull down the latest, and check out `src/__tests__/http-spec.ts` - basically sanity checking the typings against our example. There are some inconsistencies that typescript surfaces - could you take a look and either fix the typings or fix the example so that there are no more type errors?

**Comment by XVincentX** [comment]
I'll check it in a while — I did the same an hour ago and I think there's just some fields named differently. CollectionFormat is probably a Style in the Http Operation. I'll check it out later.

**Comment by pytlesk4** [comment]
So when this goes live, what will we do about analyzer diffs?

Do we write a migration for nodes? If we don't, then Amadeus will see potential changes to nodes they didn't expect.

Just a question, should not hold up this change.

**Comment by marbemac** [comment]
Amadeus is still getting going in general, if we get this in (and graphite/analyzer update) in next week or so shouldn't be an issue.

File: src/http-spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
+import { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
```
Comment (lottamus) [inline src/http-spec.ts]: Do we need to add a dependency or peerdep on `json-schema@7`? Running into an issue adding `@stoplight/types@9` to `api-gen3

image

Comment (marbemac) [reply src/http-spec.ts]: Yeah - `@types/json-schema` is in devDependencies right now - it should be in dependencies.

Comment (lottamus) [reply src/http-spec.ts]: Sounds good, I'll move it over 👍

Comment (XVincentX) [reply src/http-spec.ts]: Ah gosh, this was my bad!

Comment (lottamus) [reply src/http-spec.ts]: No worries! I think I was the first to run into it but all resolved now 🎉

**Pull request merged**